lib/internal/modules/esm/initialize_import_meta.js

```diff
@@ -0,0 +1,25 @@
+const { getOptionValue } = require('internal/options');
+const experimentalImportMetaResolve =
+getOptionValue('--experimental-import-meta-resolve');
+
+function createImportMetaResolve(defaultParentUrl) {
+  return async function resolve(specifier, parentUrl = defaultParentUrl) {
+    return PromisePrototypeCatch(
+      asyncESM.ESMLoader.resolve(specifier, parentUrl),
+      (error) => (
+        error.code === 'ERR_UNSUPPORTED_DIR_IMPORT' ?
+          error.url : PromiseReject(error))
+    );
+  };
```

Comment:
Does this need the nested closure? I _think_ this could just return `asyncESM.ESMLoader.resolve` since it has the same signature

Reply:
I copied this directly from existing code, was not looking to refactor it / it looks like it special cased error for directory imports when using `import.meta.resolve` unlike real `import` resolve. I think removing the special casing would allow removing the closure, but that isn't the intent of this PR.